{
    "url": "https://api.github.com/repos/spack/spack/issues/10581",
    "repository_url": "https://api.github.com/repos/spack/spack",
    "labels_url": "https://api.github.com/repos/spack/spack/issues/10581/labels{/name}",
    "comments_url": "https://api.github.com/repos/spack/spack/issues/10581/comments",
    "events_url": "https://api.github.com/repos/spack/spack/issues/10581/events",
    "html_url": "https://github.com/spack/spack/issues/10581",
    "id": 409583887,
    "node_id": "MDU6SXNzdWU0MDk1ODM4ODc=",
    "number": 10581,
    "title": "Installation issue: petsc with python3",
    "user": {
        "login": "burgreen",
        "id": 13755293,
        "node_id": "MDQ6VXNlcjEzNzU1Mjkz",
        "avatar_url": "https://avatars.githubusercontent.com/u/13755293?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/burgreen",
        "html_url": "https://github.com/burgreen",
        "followers_url": "https://api.github.com/users/burgreen/followers",
        "following_url": "https://api.github.com/users/burgreen/following{/other_user}",
        "gists_url": "https://api.github.com/users/burgreen/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/burgreen/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/burgreen/subscriptions",
        "organizations_url": "https://api.github.com/users/burgreen/orgs",
        "repos_url": "https://api.github.com/users/burgreen/repos",
        "events_url": "https://api.github.com/users/burgreen/events{/privacy}",
        "received_events_url": "https://api.github.com/users/burgreen/received_events",
        "type": "User",
        "site_admin": false
    },
    "labels": [
        {
            "id": 446616547,
            "node_id": "MDU6TGFiZWw0NDY2MTY1NDc=",
            "url": "https://api.github.com/repos/spack/spack/labels/build-error",
            "name": "build-error",
            "color": "ff3300",
            "default": false,
            "description": null
        }
    ],
    "state": "closed",
    "locked": false,
    "assignee": null,
    "assignees": [],
    "milestone": null,
    "comments": 1,
    "created_at": "2019-02-13T01:43:57Z",
    "updated_at": "2019-03-06T13:55:21Z",
    "closed_at": "2019-03-06T13:55:21Z",
    "author_association": "NONE",
    "active_lock_reason": null,
    "body": "### Steps to reproduce the issue\r\n\r\nI manually modified the petsc/package.py to depend on python >= 3.4.\r\n\r\n```console\r\n$ spack install petsc\r\n... # ImportError: No module named site.\r\n```\r\nThe problem is the last official release of PETsc does not support Python3.\r\n\r\nHowever, a Python3 compatible version of PETsc will be released soon.\r\n\r\nIn the meantime, the current master branch of PETsc is Python3 compatible.\r\n\r\nSo, right or wrong, good practice or not, I downloaded a ZIP archive of the current PETsc git repo and used that as a repeatible, checksummable archive. \r\n\r\nThis unofficial petsc archive installed fine with Python-3.7.2. \r\n",
    "performed_via_github_app": null
}